HarClient.Tests/Content/AsyncEntryGenerationBehaviour.cs

```diff
@@ -0,0 +1,73 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace JBlam.HarClient.Tests.Content
+{
+    [TestClass]
+    public class AsyncEntryGenerationBehaviour
+    {
+        [TestMethod]
+        public void CanCancelSerialisation()
+        {
+            var arbitraryRequest = new HttpRequestMessage(HttpMethod.Get, "http://example.net");
+            var unresolvableResponse = new HttpResponseMessage() { Content = new DeferredHttpContent() };
+            var sut = new HarEntrySource(arbitraryRequest, default);
+            sut.SetResponse(unresolvableResponse);
+            var cancellationTokenSource = new CancellationTokenSource();
+            var entryTask = sut.CreateEntryAsync(cancellationTokenSource.Token);
+            if (entryTask.IsCompleted)
+                throw new TestException("Creation of entry unexpectedly resolved before content was copied");
+            cancellationTokenSource.Cancel();
+            Assert.IsTrue(entryTask.IsCompletedSuccessfully, "Entry task failed to complete after cancelling unresolvable content duplication");
+            var entry = entryTask.Result;
+            Assert.IsNotNull(entry.Request, "Cancelled HAR entry generation did not record the synchronously-available request");
+            // JB 2020-07-15: at time of writing, the implementation does not return a partially-
+            // complete Response object if the content is not available.
+            Assert.IsNull(entry.Response, "HAR entry unexpectedly has a response, even though the content is unavailable");
+        }
+
+        [TestMethod]
+        public void CanCancelSerialisationBeforeResponse()
+        {
+            var unresolvableRequest = new HttpRequestMessage(HttpMethod.Get, "http://example.net");
+            var sut = new HarEntrySource(unresolvableRequest, default);
+            var cancellationTokenSource = new CancellationTokenSource();
+            var entryTask = sut.CreateEntryAsync(cancellationTokenSource.Token);
+            cancellationTokenSource.Cancel();
+            Assert.IsTrue(entryTask.IsCompletedSuccessfully, "Entry task failed to complete after cancelling unresolvable content duplication");
+            var entry = entryTask.Result;
+            Assert.IsNotNull(entry.Request, "Cancelled HAR entry generation did not record the synchronously-available request");
+        }
+
+        [TestMethod]
+        public void CanCancelSerialisationBeforeRequest()
+        {
+            var unresolvableRequest = new HttpRequestMessage(HttpMethod.Post, "http://example.net") { Content = new DeferredHttpContent() };
+            var sut = new HarEntrySource(unresolvableRequest, default);
+            var entryTask = sut.CreateEntryAsync(new CancellationToken(true));
+            Assert.IsTrue(entryTask.IsCompletedSuccessfully, "Entry task failed to complete");
+            Assert.IsNull(entryTask.Result.Request);
+        }
+
+        [TestMethod]
+        public void WaitsForContentToResolve()
```

Comment:
Separate issue: #25.

This is informed by me having totally missed that content headers were absent, because nothing is asserting that headers are present at all.